File: projects/project_2/src/Wheel.java
import javax.media.opengl.GL2;
import javax.media.opengl.glu.GLU;
import javax.media.opengl.glu.GLUquadric;

public class Wheel {
    GLU glu;
    GLUquadric quadric;
    GL2 gl;

    public Wheel(GLU glu, GL2 gl) {
        this.glu = glu;
        quadric = glu.gluNewQuadric();
        this.gl = gl;
    }

    public int createWheelList() {
        int NUM_SPOKES = 6;

        int listId = gl.glGenLists(1);
        gl.glNewList(listId, GL2.GL_COMPILE);
        gl.glPushMatrix();
        glu.gluCylinder(quadric, 10, 10, 1, 50, 5);
        gl.glRotated(-270, 1, 0, 0);
        gl.glTranslated(0, 0.5, 0);

        for(int i = 0; i < NUM_SPOKES; i++) {
            double angle = i * 360.0 / NUM_SPOKES;
            gl.glPushMatrix();
            gl.glRotated(angle, 0, 1, 0);
            glu.gluCylinder(quadric, 0.2, 0.2, 9.9, 10, 5);
            gl.glPopMatrix();
        }

        gl.glPopMatrix();
        gl.glEndList();

        return listId;
    }
}

File: Animation/src/Wheel.cc
#ifdef __APPLE__
#include <GLUT/glut.h>
#else
#include <GL/gl.h>
#include <GL/glut.h>
#endif
#include "Wheel.h"
#include "Cylinder.h"

Wheel::Wheel()
{
    radius = 3.5;
    tire_thickness = 0.4;
}

/*-------------------------------*
 * Make a wheel for the unicycle *
 *-------------------------------*/
void Wheel::newInstance ()
{
    const int NUM_SPOKES = 5;
    /* black rubber */
    Cylinder * spoke;

    spoke = new Cylinder (.5, .5, 1);
    spoke->setColor (0.75, 0.75, 0.75);
    spoke->newInstance();

    list_id = glGenLists(1);
    glNewList (list_id, GL_COMPILE);
    glPushMatrix();
    glRotatef (90, 1, 0, 0);

    glColor3ub (70,40, 40);
    glutSolidTorus (tire_thickness, radius, 10, 30);

    /* make the spokes */
    for (int k = 0; k < NUM_SPOKES; k++)
    {
        float angle = k * 360.0 / NUM_SPOKES;
        glPushMatrix();
        glRotatef (angle, 0, 0, 1);
        glRotatef (-90, 0, 1, 0);
        glScalef (0.75 * tire_thickness, 0.75 * tire_thickness,
                  radius - 0.2);
        spoke->draw();
        glPopMatrix();
    }
    /* make the hub */
    glPushMatrix();
    glTranslatef (0, 0, -tire_thickness/2);
    glScalef (0.3 * radius, 0.3 * radius, tire_thickness);
    spoke->draw();
    glPopMatrix();
    glPopMatrix();
    glEndList();
}

